### PR TITLE
feat: Add new field non_os_package_support to ECR resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
***Issue***: https://lacework.atlassian.net/browse/ALLY-665

***Description:***
Add new field non_os_package_support for ecr terraform resource

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: examples/resource_lacework_integration_ecr/iam_role/variables.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+variable "external_id" {
+  type      = string
+  sensitive = true
+  default   = "12345"
+}
+variable "registry_domain" {
+  type      = string
+  default   = "YourAWSAccount.dkr.ecr.YourRegion.amazonaws.com"
+}
+variable "non_os_packages" {
```
Comment (vatasha) [inline examples/resource_lacework_integration_ecr/iam_role/variables.tf:19]: nit: same comment as other pr. Let's rename this variable to something more clear like `non_os_package_support`

File: integration/ecr_auth.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package integration
+
+import (
+	"encoding/json"
+	"os"
+)
+
+type ecrCredentialsFile struct {
```
Comment (vatasha) [inline integration/ecr_auth.go:8]: nit: camelCase for JSON. Also why are the credential fields not private? Do they need to be accessible somewhere else?

File: integration/integration.go [hunk 3]
```diff
@@ -57,6 +57,18 @@ func GetIntegrationName(result string) string {
 	return res.Data[0].Name
 }
 
+func GetEcrWithCrossAccountCreds(result string) api.AwsEcrWithCrossAccountIntegration {
```
Comment (vatasha) [inline integration/integration.go:60]: Same comment as before, concerned about how we're splitting the string here.

**Review by vatasha — CHANGES_REQUESTED** [review]
I want to to talk more about type assertions and splitting of strings.

**Review by vatasha — APPROVED** [review]
LGTM

File: integration/ecr_auth.go (hunk 2 above)
Comment (afiune) [reply integration/ecr_auth.go:8]: They have to be public to be able to unmarshal them.

File: integration/resource_lacework_integration_ecr_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	awsCreds, err := ecrLoadDefaultCredentials()
+	if assert.Nil(t, err, "this test requires you to set AWS_ECR_IAM environment variable") {
+		terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: "../examples/resource_lacework_integration_ecr/iam_role",
+			Vars: map[string]interface{}{
+				"integration_name": "Amazon Elastic Container Registry Example",
+				"role_arn":         awsCreds.RoleArn,
+				"external_id":      awsCreds.ExternalID,
+				"registry_domain":  awsCreds.RegistryDomain,
+				"non_os_package_support":  false,
```
Comment (afiune) [inline integration/resource_lacework_integration_ecr_test.go:25]: We need to format this file.

**Review by afiune — APPROVED** [review]
tenor-218294250

**Pull request merged**